src/components/views/elements/PowerSelector.tsx

```diff
@@ -106,27 +106,35 @@ export default class PowerSelector<K extends undefined | string> extends React.C
         });
     }
 
-    private onSelectChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
+    private onSelectChange = async (event: React.ChangeEvent<HTMLSelectElement>): Promise<void> => {
         const isCustom = event.target.value === CUSTOM_VALUE;
         if (isCustom) {
             this.setState({ custom: true });
         } else {
             const powerLevel = parseInt(event.target.value);
-            this.props.onChange(powerLevel, this.props.powerLevelKey);
             this.setState({ selectValue: powerLevel });
+            try {
+                await this.props.onChange(powerLevel, this.props.powerLevelKey);
+            } catch {
+                this.initStateFromProps();
+            }
```

Comment:
won't this blow up if the component is unmounted while the callback happens?

Reply:
Nope, you'll get a warning about setState called on unmounted component, and even that warning is going away in React 18

Reply:
when I said "blow up" I really meant "add to the console warning spam", but TIL about react 18.

Slightly concerned that we'll be on React 17 for a while yet (React 18 was released 2 years ago?) and in the meantime this stuff does add to the red light fatigue in the console, but I guess it's not worth complicating the code to deal with.

Reply:
Done.

Reply:
oh, you fixed it anyway.